src/routes/settings/+page.svelte

```diff
@@ -0,0 +1,320 @@
+    );
+
+    const getTranslationPairsLanguages = (translationPairs: TranslationPair[]) => {
+        translationPairsLanguages = Array.from(
+            new Set(translationPairs.map((translationPair) => translationPair.languageId))
+        ).map((tpl) => ({
+            languageId: tpl,
+            englishDisplay: data.languages.find((l) => l.id === tpl)?.englishDisplay ?? '',
+        }));
+
+        if ($searchParams.currentLanguageId === 0) {
+            $searchParams.currentLanguageId = translationPairsLanguages[0]?.languageId ?? 0;
+        }
+    };
+
+    const openDeleteTranslationPair = (translationPairId: number) => () => {
+        currentDeleteTranslationPairId = translationPairId;
+        openDeleteModal = true;
+    };
+
+    const deleteTranslationPair = async () => {
+        try {
+            await deleteToApi(`/translation-pairs/${currentDeleteTranslationPairId}`);
+            translationPairs = translationPairs.filter((tp) => tp.translationPairId !== currentDeleteTranslationPairId);
+        } catch (e) {
+            errorMessage = 'An error occurred while deleting the translation pair.';
+            openErrorModal = true;
+            currentDeleteTranslationPairId = 0;
+            isTransacting = false;
+            log.exception(e);
+        } finally {
+            currentDeleteTranslationPairId = 0;
+            isTransacting = false;
+        }
+    };
+
+    const addTranslationPair = async () => {
+        isTransacting = true;
+        openCreateModal = false;
+
+        try {
+            const response = await postToApi<NewTranslationPairResponse>(`/translation-pairs`, {
+                languageId: $searchParams.currentLanguageId,
+                key: newKey,
+                value: newValue,
+            });
+            if (response) {
+                translationPairs = [
+                    ...translationPairs,
+                    {
+                        translationPairId: response.id,
+                        languageId: $searchParams.currentLanguageId,
+                        translationPairKey: newKey,
+                        translationPairValue: newValue,
+                        languageEnglishDisplay: currentLanguageDisplayname,
+                    },
+                ];
+                translationPairs = translationPairs.sort((a, b) =>
+                    a.translationPairKey.localeCompare(b.translationPairKey)
+                );
+            }
+        } catch (e) {
+            errorMessage = 'An error occurred while adding the translation pair.';
+            let containsErrorMessage = isApiErrorWithMessage(e, 'Key already exists for this language');
+
+            if (containsErrorMessage) {
+                errorMessage = 'Key already exists for this language.';
+            }
+            openErrorModal = true;
+            newKey = '';
+            newValue = '';
+            isTransacting = false;
+            log.exception(e);
```

Comment:
What do you think about only doing log.exception `if (!containsErrorMessage)` that way we don't get alerts for the expected error of adding an existing key?